### ISSUE TITLE
Bug: APOC doesn't read configuration file for Neo4j executable installs

### ISSUE THREAD

**Issue by InverseFalcon** [neo4j-contrib/neo4j-apoc-procedures#313]
This affects the community versions for Mac and Windows, when using the DMG or EXE installations. The problem does not occur with the Mac tar/gz install, and presumably not on the Windows zip install.

On the executable installations, with APOC installed (so far confirmed for 3.1.x versions of APOC), APOC does not read the config file used by the executable installations. 

This can be confirmed by calling `apoc.config.map()` or `apoc.config.list()`, both show up as empty. 

Properties used by Neo4j itself can be set and configured normally using the config files, so the issue is limited to APOC's reading of the files only.

This is likely the root issue behind bugs #294 and #312 .

**Comment by danielamigos** [comment]
I can confirm that the Windows zip install is not affected by this issue.

**Comment by kathyqian** [comment]
I can confirm that the Mac tar install doesn't have this issue either

**Comment by jexp** [comment]
@neo4j-contrib/larus Please timebox this to 2-3 hours of investigation, then I'll check.

**Comment by jexp** [comment]
I made some progress @InverseFalcon  it seems that on OSX at least the kernel extension factory which also reads the config is not started properly if you put the jar into the `default.graphdb/plugins` directory, there is a separate directory which is included in the classpath in info.plist.

But you somehow have to put it in both :(

* OSX: `/Applications/Neo4j\ Community\ Edition\ <version>.app/Contents/Resources/app/plugins` AND `/Users/<user>/Documents/Neo4j/default.graphdb/plugins`

not sure what it is on windows

the relevant section from info.plist is

 ```<key>ClassPath</key>
<string>$APP_PACKAGE/Contents/Resources/app/.install4j/i4jruntime.jar:$APP_PACKAGE/Contents/Resources/app/bin/neo4j-desktop-3.1.3.jar:$APP_PACKAGE/Contents/Resources/app/plugins/*</string><!-- I4J_INSERT_CLASSPATH -->```

**Comment by AngeloBusato** [comment]
@jexp in windows is the same, if we put the jar into `C:\Users\<user>\Documents\Neo4j\default.graphdb\plugins` AND into `C:\Program Files\Neo4j CE 3.1.3\plugins` it works. 
If we remove the jar from  'C:\Users\<user>\Documents\Neo4j\default.graphdb\plugins' and so it is only into the install directory (`C:\Program Files\Neo4j CE 3.1.3\plugins`) it works the same.

**Comment by jexp** [comment]
@AngeloBusato When I tried it on the mac, I had to put in both.

If I only put it in `/Applications/Neo4j\ Community\ Edition\ <version>.app/Contents/Resources/app/plugins` it didn't install the procedures.

The problem is that the Desktop-App in the "Options" pane points to the "plugins" location within the graphdb directory, which works in general just not for the ExtensionFactory.

So for the time being we'll have to document it in the readme (best with pictures) and make sure that bit makes it into the main docs, best with a separate include-tag.
